### PR TITLE
Add setting to delete past calendar events

### PR DESCRIPTION
**Pull request by yabirgb**
Adds a new setting to delete past calendar events. By default is set to "do it". Also per event this configuration can be set.

Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/tests/unit/test_tasks_manager.py [hunk 1]
```diff
@@ -761,10 +763,80 @@ def test_send_ws_calendar_reminder(
...
+            address=None,
+            blockchain=None,
+            color=None,
+            should_delete=True,
+        ),
+    )
+    calendar_db.create_calendar_entry(
+        calendar=CalendarEntry(
+            name='ENS renewal',
+            description='renew hania.eth',
```
Comment (LefterisJP) [inline rotkehlchen/tests/unit/test_tasks_manager.py:808]: awwwww but that's like .... really in the future (in the real world)

**Review by LefterisJP — COMMENTED** [review]
Left some comments. Also some tests failed and are related to the changes here.

File: rotkehlchen/db/calendar.py [hunk 2]
```diff
@@ -42,6 +43,7 @@ def serialize(self) -> dict[str, Any]:
             'name': self.name,
             'description': self.description,
             'timestamp': self.timestamp,
+            'should_delete': self.should_delete,
```
[the remark this answers is not in the document]
Comment (yabirgb) [reply rotkehlchen/db/calendar.py:46]: this is not the db but the api. Will change in the db insertion

**Review by LefterisJP — COMMENTED** [review]
Changes look okay but at least 1 relevant test failed.

`rotkehlchen/tests/unit/test_tasks_manager.py::test_send_ws_calendar_reminder`

**Review by LefterisJP — APPROVED** [review]
LGTM. Failed test is a 503 from poloniex.

**Pull request merged**